python/cudf/cudf/core/reshape.py

```diff
@@ -391,6 +457,12 @@ def concat(objs, axis=0, join="outer", ignore_index=False, sort=None):
         return df
 
     # If we get here, we are always concatenating along axis 0 (the rows).
+    if keys is not None:
+        raise NotImplementedError(
+            "Concatenation along axis = 0 "
+            "when passing a dictionary is not supported yet."
+        )
+
```

Comment:
*suggestion*: As above, I think if we error-check for valid input up front, we can elide the check here.

Reply:
done